src/pilot/assign_role.py

```diff
@@ -1094,19 +1120,27 @@ def generate_osd_config_with_journals(controllers, osd_drives, ssds):
     return osd_config
 
 
-def get_pci_bus_number(spinner, controllers):
-    bus = None
+def get_by_path_device_name(physical_disk, controllers):
+    disk_index = physical_disk.description.split(" ")[1]
     for controller in controllers:
-        if controller.id in spinner.id:
-            bus = controller.bus.lower()
-            break
-    return bus
-
-
-def get_by_path_device_name(pci_bus_number, physical_disk):
-    return ('/dev/disk/by-path/pci-0000:'
-            '{pci_bus_number}:00.0-sas-0x{sas_address}-lun-0').format(
-                pci_bus_number=pci_bus_number,
+        if physical_disk.controller == controller.id and \
+                controller.model.startswith("PERC H740P"):
+            return ('/dev/disk/by-path/pci-0000:'
+                    '{pci_bus_number}:00.0-scsi-0:2:{disk_index}:0').format(
+                pci_bus_number=controller.bus.lower(),
```

Comment:
It would be nice to create a get_pci_bus_number() method that takes a single controller and returns controller.bus.lower().  We could then change the 3 locations here to call that method.

Reply:
I have created new method get_pci_bus_number() as suggested.